SeleniumSolarTechnology/SpecFlowSeleniumSolarTechnology/StepDefinitions/FilteringProductsStepDefinitions.cs

```diff
@@ -0,0 +1,65 @@
+using NUnit.Framework;
+using SpecFlowSeleniumSolarTechnology.Pages;
+using System;
+using TechTalk.SpecFlow;
+
+namespace SpecFlowSeleniumSolarTechnology.StepDefinitions
+{
+    [Binding]
+    public class FilteringProductsStepDefinitions : UITestFixture
+    {
+        public FilteringProductsStepDefinitions(ScenarioContext scenarioContext) : base(scenarioContext)
+        {
+            Console.WriteLine("FilteringProductsStepDefinitions(ScenarioContext scenarioContext) : base(scenarioContext)");
+        }
+
+        [Given(@"Open Solartechnology Shop page")]
+        public void GivenOpenSolartechnologyShopPage()
+        {
+            homePage.Open();
+        }
+
+        [Given(@"Open Solar Panels link")]
+        public void GivenOpenSolarPanelsLink()
+        {
+            homePage.OpenSolarPanels();
+        }
+
+        [When(@"Count product items before")]
+        public void WhenCountProductItemsBefore()
+        {
+            int countProductsBefore = catalogPage.CountProductItems();
+            ScenarioContext.Current["CountBefore"] = countProductsBefore;
+            Console.WriteLine($"Products before filtering: {countProductsBefore}");
+        }
+
+        [When(@"Open filter by brands")]
+        public void WhenOpenFilterByBrands()
+        {
+            catalogPage.OpenFilters();
+        }
+
+        [When(@"Check Brand '([^']*)'")]
+        public void WhenCheckBrand(string p0)
+        {
+            catalogPage.CheckBrand("JA Solar");
```

Comment:
this should be variable name but not "JA Solar"
Rename p0 to something like brandName